test/groovy/tests/TestGenerateAndExecutePipeline.groovy

```diff
@@ -181,6 +192,29 @@ class TestGenerateAndExecutePipeline extends PipelineStepSpecification {
         1 * mockLeaveLock()
     }
 
+    @Unroll
+    def '_isReleaseOrPr returns #isReleaseOrPr when isRelease returns #isReleaseValue and _isPr returns #isPr'() {
+
+        given:
+        def mockIsRelease = createIsReleaseMock()
+        def mockIsPr      = createIsPrMock()
+        when:
```

Comment:
It is posible to add some comments besides when:
example: when: "the script is release or the PR is created"

Reply:
interesting idea. It doesn't conform to the spock conventions that I am familiar with and I think the spec name/description should capture that. Have you done that in your spock tests?

Reply:
You can put arbitrary data into tables....but codenarc may complain that you're creating a variable that isn't used.....

Reply:
@YMARTINEZG is `_isReleaseOrPr` not clear enough?

Reply:
> You can put arbitrary data into tables....but codenarc may complain that you're creating a variable that isn't used.....

I was wrong, codenarc did _not_ complain. Good for specs, bad for our code linting. This means any of our `where`s could have extraneous data.

Reply:
wrt, the [this orphaned comment](https://github.com/gitmlynch/pr_exercise/pull/1#discussion_r298231451), I'm not crazy about the idea of adding test data to the `when:` block that only gets used in the spec name.

Reply:
still an interesting idea though